File: ObjectiveC/FlexGrid101/SharedFilterData.m
//
//  SharedFilterData.m
//  FlexGrid101
//
//  Created by Evangelists on 7/28/15.
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "SharedFilterData.h"

@implementation SharedFilterData
{
    SharedFilterData *moreSharedData;
}
@synthesize filterSet;
@synthesize filterOperation;
@synthesize filterString;

+(SharedFilterData *)sharedInstance{
    static SharedFilterData *sharedInstance = nil;
    
    @synchronized(self){
        if(!sharedInstance){
            sharedInstance = [[SharedFilterData alloc] init];
        }
    }
    return sharedInstance;
}

@end

File: Swift/FlexGrid101/FlexGrid101/SharedFilterData.swift
//
//  SharedFilterData.swift
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import Foundation

class SharedFilterData {
    class var sharedInstance: SharedFilterData {
        struct Static {
            static var token: dispatch_once_t = 0
            static var instance: SharedFilterData? = nil
        }

        dispatch_once(&Static.token){
            Static.instance = SharedFilterData()
        }
        return Static.instance!
    }
    var filterSet: Bool!
    var filterOperation: Int!
    var filterString: String!
}
